### PR TITLE
[QTM-582] Prepares SSR-Simulator for Unit Tests

### PR DESCRIPTION
**Pull request by alizeleal**
Description: https://jirasoftware.catho.com.br/browse/QTM-582

## Review guide
- [ ] Unit tests
- [ ] Code review

### Browsers review
- [ ] Layout review

### CODE REVIEW

File: components/ScrollUpButton/ScrollUpButton.unit.test.jsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/**
+ * @jest-environment jsdom
+ */
+
```
Comment (MarcosViniciusPC) [inline components/ScrollUpButton/ScrollUpButton.unit.test.jsx]:
```suggestion
```

File: package.json [hunk 2]
```diff
@@ -54,7 +59,9 @@
     "lint:all": "yarn lint:components && yarn lint:js && yarn lint:pages",
     "lint": "eslint --ext js,jsx --quiet",
     "format": "prettier --write ./**/*.{js,jsx,json}",
-    "prepush": "yarn lint:all"
+    "prepush": "yarn lint:all",
+    "test": "jest",
+    "test:components": "test ./components"
```
Comment (MarcosViniciusPC) [inline package.json]:
```suggestion
    "test:components": "yarn test ./components"
```

File: package.json [hunk 3]
```diff
@@ -39,6 +41,9 @@
     "eslint-plugin-jsx-a11y": "^6.1.2",
     "eslint-plugin-prettier": "^3.0.1",
     "eslint-plugin-react": "^7.12.3",
+    "jest": "^29.7.0",
+    "jest-css-modules": "^2.1.0",
```
Comment (MarcosViniciusPC) [inline package.json]:
```suggestion
```

**Review by MarcosViniciusPC — CHANGES_REQUESTED** [review]
[] Code review
[] Test

The tests didn't work here :(. Put "yarn" inside the "test:components" script to validate.

You need to configure Jest to test components that import .css and other files. In this case, it is necessary:

Create mock files in the project root called "jest-config.js", "styleMock" and "fileMock.js", similar to those in Quantum.

Create a prop called "jest" in ssr-simulator's package.json, also similar to Quantum.

Comments:

The "jest-config.js" file created in ssr-simulator should only have "import '@testing-library/jest-dom'".

Check the "Handling Static Assets" and "Mocking CSS Modules" sections of the doc https://jestjs.io/docs/webpack for more details on the above suggestions.


In addition to the points mentioned above, check the suggestions below.

File: package.json [hunk 4]
```diff
@@ -54,12 +58,27 @@
     "lint:all": "yarn lint:components && yarn lint:js && yarn lint:pages",
     "lint": "eslint --ext js,jsx --quiet",
     "format": "prettier --write ./**/*.{js,jsx,json}",
-    "prepush": "yarn lint:all"
+    "prepush": "yarn lint:all",
+    "test": "jest",
+    "test:components": "yarn test ./components"
```
Comment (MarcosViniciusPC) [inline package.json:63]: Why not run "yarn jest..." directly instead of calling "yarn test..."?

**Review by MarcosViniciusPC — APPROVED** [review]
[x] code review
[x] test

File: package.json (hunk 4 above)
Comment (alizeleal) [reply package.json:63]: I'm following CC and Quantum's model ;)

**Pull request merged**